test/jdk/jdk/jfr/api/recording/event/TestShortPeriod.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.jfr.api.recording.event;
+
+import java.time.Duration;
+
+import jdk.jfr.Event;
+import jdk.jfr.FlightRecorder;
+import jdk.jfr.consumer.RecordingStream;
+import jdk.test.lib.jfr.EventNames;
+
+/**
+ * @test Tests that periodic events are not disabled when using a very short
+ *       period
+ * @key jfr
+ * @requires vm.hasJFR
+ * @library /test/lib /test/jdk
+ * @run main/othervm jdk.jfr.api.recording.event.TestShortPeriod
+ */
+public class TestShortPeriod {
+
+    static class PeriodicEvent extends Event {
+    }
+
+    private static Thread startLoopingThread() {
+        var t = new Thread(() -> {
+            while (!Thread.interrupted()) {}
+        });
+        t.start();
+        return t;
+    }
+
+    public static void main(String... args) {
+        testNativeEventPeriod();
+        testJavaEventPeriod();
+        testExecutionSamplePeriod();
+    }
+
+    private static void testNativeEventPeriod() {
+        try (var r = new RecordingStream()) {
+            r.enable(EventNames.JVMInformation).withPeriod(Duration.ofNanos(1));
+            r.onEvent(e -> r.close());
```

Comment:
What happens when the event is, in fact, disabled?
The test will run forever, right?

Could you rather use something like `CountDownLatch` which will get decremented in `onEvent()` handler and the test method would wait for that latch for at most N milliseconds - failing when the wait times out?

Reply:
I think it's better to use the timeout provided by jtreg than hardwiring in a number in the test.

The test could be run on different hardware or in modes that are very slow.  It's better to allow the framework to provide the timeout. The test should not fail, so it shouldn't be an issue.

I think the code looks fine, but could you remove the update to the documentation / specification. Also, perhaps add a comment in the test that the execution event requires a separate thread because the thread streaming events gets excluded from JFR to avoid recursion.

I will take it for a spin a few hundred times on different platforms to see if there are any intermittent issues.

Reply:
I ran the test about 200 times on Windows, Linux and Mac and it seems fine.